src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkTests.ps1

```diff
@@ -60,6 +60,15 @@ function Test-VirtualNetworkCRUD
         Assert-AreEqual $subnetName $list[0].Subnets[0].Name
         Assert-AreEqual "10.0.1.0/24" $list[0].Subnets[0].AddressPrefix
         Assert-AreEqual $expected.Etag $list[0].Etag
+
+        # Test virtual network private ip address - available
+        $testResponse1 = Get-AzureRmvirtualNetwork -Name $vnetName -ResourceGroupName $rgname | Test-AzureRmPrivateIpAddressAvailability -IpAddress "10.0.1.10"
+        Assert-AreEqual true $testResponse1.Available
+
+		# Test virtual network private ip address - not available
+        $testResponse2 = Get-AzureRmvirtualNetwork -Name $vnetName -ResourceGroupName $rgname | Test-AzureRmPrivateIpAddressAvailability -IpAddress "10.0.1.3"
+        Assert-AreEqual false $testResponse2.Available
+		Assert-AreEqual 5 @($testResponse2.AvailableIpAddresses).Count
```

Comment:
Shouldn't make much difference to check in use nic or reserved ip address. Will leave this as is.